### PR TITLE
Draft: Add 008 network tests

### PR DESCRIPTION
**Pull request by rvem**
## Related issue(s)

https://issues.serokell.io/issue/TM-491

#### Related changes (conditional)

- Tests
  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

[//]: # (Add more docs here if you have them in the repository)
- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)
    - Haddock

#### Stylistic guide (mandatory)

- [x] My commits comply with [the following policy](https://www.notion.so/serokell/Commit-and-PR-policy-4cf98e1a910a415d86b5f2491d9af1af).
- [x] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -62,14 +62,21 @@ steps:
...
       ./result/bin/baseDAO-test --nettest-run-network
-      -A localhost -P 8735
+
+  - label: test-local-chain-008
+    env:
+      TASTY_NETTEST_NODE_ENDPOINT: "http://localhost:8734"
+    if: *not_scheduled
+    commands: *test
 
   - label: ligo-test
```
Comment (pasqu4le) [inline .buildkite/pipeline.yml:81]: Please update ligo tests to work on both chains as well, an open PR should be merged soon to run the tests on Nettest as well.

Also, judging from CI, you also forgot to update `callFrom` in the Haskell module of the LIGO version.

Comment (rvem) [reply .buildkite/pipeline.yml:81]: Right, I've missed that ligo part is covered by a separate haskell project :confused:

**Review by pasqu4le — COMMENTED** [review]
:ok_hand: aside from the failing tests it LGTM

File: ligo/haskell/test/Test/Ligo/BaseDAO/Management.hs [hunk 2]
```diff
@@ -57,9 +56,9 @@ test_BaseDAO_Management =
   [ testGroup "Ownership transfer"
     [ nettestScenarioCaps "Contract forbids XTZ transfer" $
         withOriginated 2 (\(owner:_) -> initialStorage owner) $ \[owner, wallet1] baseDao ->
+          withSender (AddressResolved owner) $
           transfer TransferData
```
Comment (pasqu4le) [inline ligo/haskell/test/Test/Ligo/BaseDAO/Management.hs:60]:
```suggestion
          withSender (AddressResolved owner) $ transfer TransferData
```

File: .buildkite/pipeline.yml [hunk 3]
```diff
@@ -62,23 +62,47 @@ steps:
...
+  # TODO [#124] Uncomment
+  # - label: ligo-test-local-chain-007
+  #   env:
+  #     TASTY_NETTEST_NODE_ENDPOINT: "http://localhost:8735"
+  #   if: *not_scheduled
+  #   commands: &ligo-test
+  #   - nix-build ci.nix -A packages.baseDAO-ligo-meta.tests.baseDAO-test
+  #   # Running only on emulator for now due to oversized operations
+  #   - nix run -f ci.nix tezos-client -c
+  #     ./result/bin/baseDAO-test --nettest-run-network
```
Comment (pasqu4le) [inline .buildkite/pipeline.yml:99]: Couldn't we have these un-commented but with the `--nettest-no-run-network` option instead?

Comment (rvem) [reply .buildkite/pipeline.yml:99]: there is such a job above

Comment (pasqu4le) [reply .buildkite/pipeline.yml:99]: Wait, above there's the ones for the haskell version, aka `test-local-chain-007` and `test-local-chain-008` and one for the ligo version `ligo-test`.

My suggestion is to (aside from keeping the haskell version ones) to remove `ligo-test` and add `ligo-test-local-chain-007` and `ligo-test-local-chain-008` but with `--nettest-no-run-network` (and a `TODO [#124] change to nettest-run-network` just above that)

File: .buildkite/pipeline.yml [hunk 4]
```diff
@@ -62,23 +62,47 @@ steps:
...
-  - label: test
+  - label: test-local-chain-007
+    env:
+      TASTY_NETTEST_NODE_ENDPOINT: "http://localhost:8735"
     if: *not_scheduled
-    commands:
+    commands: &test
     - nix-build ci.nix -A packages.baseDAO.tests.baseDAO-test
-    # Emulator + local chain for Delphi protocol
+    - export TASTY_NETTEST_DATA_DIR="$(mktemp -d --tmpdir="$PWD")"
```
Comment (pasqu4le) [inline .buildkite/pipeline.yml:71]: `$PWD` :suspect:

File: .buildkite/pipeline.yml (hunk 3 above)
Comment (rvem) [reply .buildkite/pipeline.yml:99]: Hmm, what is the reason to have two jobs with `--nettest-no-run-network`? They'll run the same set of tests twice :thinking:

File: .buildkite/pipeline.yml (hunk 4 above)
Comment (rvem) [reply .buildkite/pipeline.yml:71]: oh, right, here it needs to be updated as well

File: .buildkite/pipeline.yml (hunk 3 above)
Comment (pasqu4le) [reply .buildkite/pipeline.yml:99]: That's a fair point, I just want to make sure not to leave `ligo-test` behind.

How about replacing it with `ligo-test-local-chain-007` (`--nettest-no-run-network` with comment) and keep `ligo-test-local-chain-008` only fully commented out?

Comment (rvem) [reply .buildkite/pipeline.yml:99]: Well, Ideally we should have 3 jobs:
1) _only_ emulator tests
2) _only_ 007 local-chain tests
3) _only_ 008 local-chain tests
This way it's currently done in `morley` repository

But I don't know how to do it in buildkite yet:(
See [this thread](https://forum.buildkite.community/t/escaped-shell-variable-in-pipeline-config/1451) on buildkite forum

Comment (rvem) [reply .buildkite/pipeline.yml:99]: I agree, let's leave only `ligo-test-local-chain-007` for now with a comment that it doesn't actually run any network tests

Comment (rvem) [reply .buildkite/pipeline.yml]: @pasqu4le, I've refactored test jobs once again to have the :proper: structure. Please, rereview once again:)

**Review by pasqu4le — APPROVED** [review]
LGTM now :ok_hand:

**Pull request merged**